feat(trends): Implicitly set trend parameter as filter value in query

udameli opened the pull request:
For the performance trends view, add current trend parameter field value as an additional condition filter to the query

udameli commented on static/app/views/performance/data.tsx:
```diff
@@ -456,6 +457,10 @@ function generateGenericPerformanceEventView(
 
+  if (isTrends) {
+    eventView.additionalConditions.addFilterValues('has', ['measurements.vital']);
+  }
```
`has:measurements.vital` should only be added to the trends query (that was the assumption i made)

Zylphrex replied:
`measurements.vitals` is not an actual measurement. In this case, the measurement we want to add depends on the parameter dropdown. So if `LCP` was selected, you'll want to add `has:measurements.lcp` and so on. Try doing this manually in the search bar to see how that affects the results.

![image](https://user-images.githubusercontent.com/10239353/149386167-48038d31-8de1-445e-b06b-d92960013c0b.png)

udameli replied:
sorry my initial understanding of the task wasn't accurate. Updated the approach

k-fish commented:
Needs a few tweaks, discussed offline 😄

Commit:
feat(trends): Implicitly set trend parameters as filter values in the query

For the performance trends view, check if current parameter is a web vital parameter. If so, add its value as an additional condition filter to the query

Fixes VIS-1273

udameli commented:
> Needs a few tweaks, discussed offline 😄

done!

k-fish reviewed (commented):
Looks great 👏 , we should probably include a test here to make sure this implicit condition is making it to the api request.

I'd probably add or modify one of the tests in `tests/js/spec/views/performance/trends/index.spec.jsx`

Commit:
test(trends): Test web vital parameters being added as additional condition filters to query

k-fish commented on tests/js/spec/views/performance/trends/index.spec.jsx:
```diff
@@ -410,6 +411,56 @@ describe('Performance > Trends', function () {
     }
   }, 10000);
 
+  it('choosing a web vitals parameter adds it as an additional condition to the query', async function () {
```
👏

k-fish reviewed (approved):
Looks great, 🚢  it!
